src/components/attribute-picker/attribute-picker.js

```diff
@@ -167,6 +173,12 @@ class AttributePicker extends ArrowKeysMixin(RtlMixin(LocalizeLabsElement(LitEle
 		this._inputFocused = false;
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this.addEventListener('focusin', this._onFocusIn);
+		this.addEventListener('focusout', this._onFocusOut);
```

Comment:
The recommended approach for this kind of thing is to wire event listeners up in `connectedCallback` and remove them in `disconnectedCallback`. Otherwise if this element gets removed from the DOM the listeners are still there, and then if it gets re-added it'll add a second set of listeners.

Otherwise you could keep the listeners on the `<div>` and then Lit will manage them for you.

Reply:
I'd put them in `firstUpdated` based on how the [`tag-list-item-mixin`](https://github.com/BrightspaceUI/core/blob/main/components/tag-list/tag-list-item-mixin.js#L187) did it, but moving them to connected & disconnected also works, so I'll make that switch! I'd initially had it tied to a div, but it had problems keeping the focus when switching between items (even though they were inside), so moving it to here resolved the issue.

Reply:
Clearly nobody's perfect haha

Reply:
We do have this [documented](https://daylight.d2l.dev/developing/best-practices/events/#originating-from-the-host-element) in Daylight as recommending using `firstUpdated` for cases like this so if our recommendation is changing here we should keep that consistent.

Reply:
Unresolved this thread, as Margaree is correct that the way you had it initially is the correct approach.

Reply:
No problem! At least we know what the official recommendation is now 😄

Reply:
Haha I'm pretty sure I've made this mixup a couple times, hence the doc that got written. Maybe this time it'll stick!